src/Domain/Entity/Promotion.php

```diff
@@ -28,22 +30,29 @@ class Promotion
     /** @var bool */
     private $isSuperPromotion;
 
+    /** @var RelatedLink[]|null */
+    private $relatedLinks;
```

Comment:
Ahh, so we can check later.... How come they aren't of type `UnfetchedRelatedLinks`?

Reply:
When dealing with a property that is an array of things rather than a single entity: null corresponds to "unfetched -  i haven't asked for this data" and `[]` is "I requested this data but there was nothing there".

(contrast dealing with a property that corresponds to a single entity where an Unfetched* entity is "unfetched -  i haven't asked for this data" and null is "I requested this data but there was nothing there")

Reply:
> How come they aren't of type `UnfetchedRelatedLinks`

We can't typehint on "array or UnfetchedRelatedLinks", but we can do `?array`